src/main/kotlin/apply/application/mail/MailService.kt

```diff
@@ -102,15 +106,23 @@ class MailService(
                 .onFailure { failed.addAll(addresses) }
         }
 
-        mailHistoryRepository.save(
-            MailHistory(
-                request.subject,
-                request.body,
-                request.sender,
-                request.recipients,
-                request.sentTime
-            )
-        )
+        saveMailHistories(mailMessage.id, succeeded, failed)
+    }
+
+    fun sendMailsByBccSynchronous(request: MailData, files: Map<String, ByteArrayResource>) {
```

Comment:
1. 단건으로 보내는 거라 s는 빼는게 맞겠네요.

```suggestion
    fun sendMailByBccSynchronous(request: MailData, files: Map<String, ByteArrayResource>) {
```

2. files도 MailData에 합치는게 좋을 것 같은데 스프링 리소스를 쓰고 있어서 따로 분리했나보네요. 저는 인터페이스 추가해서 하나로 합치는게 나을 것 같은데 다른 분들 어떠신가요. 따로 다루는게 나을까요?

3. 첨부 파일이 없는 경우가 많을거라 기본값으로 emptyMap()을 설정하는게 좋을 것 같습니다.

Reply:
이번에는 메일 예약에서 파일 첨부 기능을 지원하지 않기로 해서 크게 신경쓰지는 않았는데, 장기적인 관점으로는 구구 말씀처럼 합치는 방향으로 가는게 좋아보여요 ㅎㅎ